nucypher/blockchain/eth/actors.py

```diff
@@ -763,17 +763,53 @@ def _verify_active_ritual(self, decryption_request: ThresholdDecryptionRequest):
                 f"Node not part of ritual {decryption_request.ritual_id}",
             )
 
-    def _verify_ciphertext_authorization(
+    def _verify_encryption_authorization(
         self, decryption_request: ThresholdDecryptionRequest
     ) -> None:
-        """check that the ciphertext is authorized for this ritual"""
+        """Check that the encryption is authorized for this ritual"""
         ciphertext_header = decryption_request.ciphertext_header
         authorization = decryption_request.acp.authorization
-        if not self.coordinator_agent.is_encryption_authorized(
-            ritual_id=decryption_request.ritual_id,
-            evidence=authorization,
-            ciphertext_header=bytes(ciphertext_header),
-        ):
+        ritual = self._resolve_ritual(decryption_request.ritual_id)
+        abi = """[
```

Comment:
I assume that you already have plans to relocate this for readability?

Reply:
Yeah absolutely. In fact, this was actually bait for you 🤠 It desperately needs a grander idea for creating agents out of known ABIs but with addresses known at runtime. This will be useful for instantiating contracts that have a known interface or are generated by a factory. As you can see here, I just hardcoded the abi and used web3py directly for the contract, but an agent-like entity would be best.

Reply:
I knew something was bait-like about this :-). 

Yes - I clearly see the need for such an interface.  I have a few ideas about how to build this I'll try to whip something up.

Reply:
Opened https://github.com/nucypher/nucypher/issues/3503